Use experimental xFullText and xOngoing search parameters in Hobbies

nikomakela opened the pull request:
HH-371.

Depends on https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/771.

Adds new experimental `xFullText` and `xOngoing` search parameters to the common components package and Hobbies app.

nikomakela committed:
feat(packages): update schema with exp search params

HH-371.

Update GraphQL schema related to events list query to include
experimental search parameters to improve full text search for ongoing
events.

NOTE: The schema also includes some new updated CMS schema changes.
Those changes are unrelated, but should not break code.

nikomakela committed:
feat(packages): change search text-parameter

HH-371.

Use "xFullText" as a (default) event search parameter instead of "text".

nikomakela committed:
feat(hobbies): use experimental xFullText search parameter

HH-371.

Change the old "text"-parameter for search to new experimental
"xFullText". Also change "allOngoing" search filter to new experimental
"xOngoing".

terovirtanen commented:
**Federation Router branch is deployed to platta: https://events-graphql-federation-sports-pr770.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
refactor(hobbies): advanced search form as compound component

HH-371.

Chop old unclear implementation of `AdvancedSearch`-component to a new
compound component, that has the search context provided with React
context tools, form state is initialized with a new hook and the form
section is rendered by a new `AdvancedSearchForm` child component that
uses the new context.

nikomakela committed:
fix(hobbies): use shallow transitions on search for squared UX

HH-371.

The search was slow and bouncy, which was especially visible in
development usage. The search state and query is controlled by URL
search params. The shallow transition in `router.push` navigation should
square out those issues.

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr770.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr770.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr770.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr770.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr770.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr770.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr770.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr770.dev.hel.ninja :laughing::tada::tada::tada:

karisal-anders reviewed (approved):
Otherwise ok, just fixing packages/components types with [#1](https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/770#issuecomment-2970244405) and making use of the enums with [#2](https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/770#discussion_r2145098747).

Needs testing, but that's expected for this experimental feature. Also related to the [change from localOngoingAnd/allOngoingAnd/allOngoing → xFullText + xOngoing](https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/770#discussion_r2145076705)

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr770.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr770.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr770.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr770.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr770.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr770.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr770.dev.hel.ninja** :rocket::rocket::rocket:
